src/libs/Pusher/pusher.js

```diff
@@ -258,9 +264,26 @@ function unsubscribe(channelName, eventName = '') {
 
         channel.unbind();
         socket.unsubscribe(channelName);
+
+        if (/private-user-accountID-\d/.test(channelName)) {
+            Network.refreshOfflineStatus();
+        }
     }
 }
 
+/**
+ * Do we have an active websocket connection to the Pusher service?
+ *
+ * @returns {Boolean}
+ */
+function isConnected() {
+    if (!socket) {
+        return false;
+    }
+
+    return socket.connection.state === CONST.PUSHER.STATE.CONNECTED;
```

Comment:
https://pusher.com/docs/channels/using_channels/connection/#querying-the-connection-state